master/internal/db/setup.go

```diff
@@ -85,61 +83,37 @@ func Connect(opts *config.DBConfig) (*PgDB, error) {
 	return db, nil
 }
 
-// IsNew checks to see if the database's migration tracking tables have been
-// created, and if so, if it's above version 0. It returns `false` if the current
-// version exists and is higher than zero, `true` otherwise.
-// This is not guaranteed to be accurate if the database is otherwise in a bad or
-// incomplete state. If an error is returned, the bool should be ignored.
-func IsNew(opts *config.DBConfig) (bool, error) {
-	dbURL := fmt.Sprintf(cnxTpl, opts.User, opts.Password, opts.Host, opts.Port, opts.Name)
-	dbURL += fmt.Sprintf(sslTpl, opts.SSLMode, opts.SSLRootCert)
-	pgOpts, err := makeGoPgOpts(dbURL)
-	if err != nil {
-		return false, err
-	}
-
-	pgConn := pg.Connect(pgOpts)
-	defer func() {
-		if errd := pgConn.Close(); errd != nil {
-			log.Errorf("error closing pg connection: %s", errd)
-		}
-	}()
-
-	exist, err := tablesExist(pgConn, []string{"gopg_migrations", "schema_migrations"})
+// Setup connects to the database and run any necessary migrations.
+// Takes a list of checks that run after the database is connected on new databases.
+func Setup(opts *config.DBConfig, newOnlyChecks ...func() error) (db *PgDB, isNew bool, err error) {
+	db, err = Connect(opts)
 	if err != nil {
-		return false, err
-	}
-	if !exist["gopg_migrations"] {
-		return true, nil
+		return db, false, err
```

Comment:
nit (non-blocking): is there a benefit to returning the db pointer if an error occurs?

Reply:
nope that was a mistake, should return nil